engines/router/missionctl/fiberapi/routing_strategy.go

```diff
@@ -54,18 +58,47 @@ func (r *DefaultTuringRoutingStrategy) SelectRoute(
 		fallbacks = append(fallbacks, defRoute)
 	}
 
-	//TODO to revisit when implementing grpc for experiment engine
-	if req.Protocol() == fiberProtocol.GRPC {
-		return nil, fallbacks, nil
+	var payload []byte
+	httpHeader := http.Header{}
+
+	switch req.Protocol() {
+
+	case fiberProtocol.HTTP:
+		payload = req.Payload()
+		httpHeader = req.Header()
+	case fiberProtocol.GRPC:
+		upiRequest, ok := req.(*upi.Request)
+		if !ok {
+			err := fmt.Errorf("failed to convert into UPI request")
+			log.Glob().Error(err.Error())
+			return nil, nil, err
+		}
+
+		predContext, err := request.UPIVariablesToStringMap(upiRequest.RequestProto.GetPredictionContext())
+		if err != nil {
+			log.Glob().Errorf("failed converting prediction context into string map: %s", err)
+			return nil, nil, err
+		}
+
+		payload, err = json.Marshal(predContext)
+		if err != nil {
+			log.Glob().Errorf("failed marshalling prediction context into payload: %s", err)
+			return nil, nil, err
+		}
+
+		for k, v := range req.Header() {
```

Comment:
are there other scenario where http2 -> http1 will be used? cause I cant think of any other case scenario since this is the only place we are doing conversion from proto to http call

Reply:
You are right, the more common case is copying MD to MD.